### PR TITLE
Implemented reading weather from json and putting it to object

### PR DESCRIPTION
**Pull request by Wielopol**

### CODE REVIEW

**Review by kmusial1988 — APPROVED** [review]

File: src/main/java/pl/sda/weather/controllers/ControllersLocation.java [hunk 1]
```diff
@@ -1,9 +1,108 @@
...
+import javax.xml.validation.ValidatorHandler;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.UUID;
+
+import static pl.sda.weather.Gui.showMainMenu;
+
 public class ControllersLocation {
+    ILocationService locationService = new LocationServiceImpl();
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/controllers/ControllersLocation.java:19]: should be private

File: src/main/java/pl/sda/weather/controllers/ControllersLocation.java [hunk 2]
```diff
@@ -1,9 +1,108 @@
...
+    public void addLocation() {
+
+        String idUUID = String.valueOf(UUID.randomUUID());
+
+       String longitudeAndLatitude;
+        String longitudeS;
+        String longitudeN;
+        String latitudeW;
+        String latitudeE;
+       do {
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/controllers/ControllersLocation.java:36]: would be greate to have terminal logic separated (MODEL_VIEW_CONTROLLER pattern) -> right now you have VIEW logic in CONTROLLER

File: src/main/java/pl/sda/weather/controllers/ControllersLocation.java [hunk 3]
```diff
@@ -1,9 +1,108 @@
...
+
+        showMainMenu();
+
+    }
+
+    public void displayLocation() {
+
+        List<LocationModel> list = locationService.getLocationModelFromFile();
+
+        for (LocationModel locationModel : list) {
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/controllers/ControllersLocation.java:93]: would be great to have stream here

File: src/main/java/pl/sda/weather/model/WeatherLine.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
+package pl.sda.weather.model;
+
+import java.util.Arrays;
+
+public class WeatherLine {
+    private City city;
+    private long time;
+    private MainWeather main;
+    private Wind wind;
+    private Clouds clouds;
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/model/WeatherLine.java:10]: btw. you don't have to declare all fields (event those not needed) -> you can just ignore those during deserialization

File: src/main/java/pl/sda/weather/service/ReadWeathers.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import pl.sda.weather.model.LocationModel;
+import pl.sda.weather.model.Weather;
+import pl.sda.weather.model.WeatherLine;
+
+import java.util.List;
+import java.util.Map;
+
+public interface ReadWeathers {
+
+    Map<String,Weather> mapWeather();
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/service/ReadWeathers.java]: getWeatherMap() would be better (mapWeather means - do mapping on Weather object)

File: src/test/java/pl/sda/weather/service/impl/ReadWeathersImplTest.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package pl.sda.weather.service.impl;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ReadWeathersImplTest {
+
+    @org.junit.jupiter.api.Test
+    void readWeather() {
```
Comment (MichalOrlowski) [inline src/test/java/pl/sda/weather/service/impl/ReadWeathersImplTest.java:16]: I'm waiting for the tests :)

File: src/main/java/pl/sda/weather/validation/LocationValidation.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LocationValidation {
+
+    public boolean valCordinates(String name) {
+        Pattern recipientPackage = Pattern.compile("^[0-9]{3}$");
+        Matcher n = recipientPackage.matcher(name);
+        return !name.isBlank() || n.find();
+//        return name.equals("") || n.find();
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/validation/LocationValidation.java:12]: remove not needed code

File: src/main/java/pl/sda/weather/validation/LocationValidation.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
+package pl.sda.weather.validation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LocationValidation {
+
+    public boolean valCordinates(String name) {
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/validation/LocationValidation.java:8]: please rename -> valCordinates is not good method name. 
Method name should be self-descriptive

File: src/main/java/pl/sda/weather/validation/LocationValidation.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package pl.sda.weather.validation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LocationValidation {
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/validation/LocationValidation.java:6]: how about LocationValidator?

Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/validation/LocationValidation.java:6]: please prepare tests for this class

File: src/main/java/pl/sda/weather/service/impl/ReadWeathersImpl.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+import pl.sda.weather.service.ReadWeathers;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Stream;
+
+public class ReadWeathersImpl implements ReadWeathers {
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/service/impl/ReadWeathersImpl.java:16]: please prepare tests for this class

File: src/main/java/pl/sda/weather/service/impl/ReadWeathersImpl.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    public Map<String, Weather> mapWeather() {
+        List<WeatherLine> weatherLines = readWeather();
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/service/impl/ReadWeathersImpl.java:20]: it will be a problem is there won't be Weather for Location in file

**Pull request closed without merging**